src/shogun/structure/GEMPLP.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2014 Jiaolong Xu
+ * Copyright (C) 2014 Jiaolong Xu
+ */
+
+#ifndef __GEMPLP_H__
+#define __GEMPLP_H__
+
+#include <shogun/lib/config.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/structure/FactorGraph.h>
+#include <shogun/structure/Factor.h>
+#include <shogun/structure/MAPInference.h>
+#include <shogun/lib/SGNDArray.h>
+
+#include <vector>
+#include <set>
+
+using namespace std;
+
+namespace shogun
+{
+#define IGNORE_IN_CLASSLIST
```

Comment:
You don't need any macro to avoid swig from generating wrappers for a class. This is what you mean by generating Python interface. In fact, a class has to be included in swig files (.i files under interfaces directory) to be accessible to modular interfaces. We use macros when we make a class accessible to interfaces but we do not want some of its methods to be accessible via modular interfaces (but it is not this macro, see e.g. SGVector).